### ISSUE TITLE
MetricGAN+ generator backward issue

### ISSUE THREAD

**Issue by ChangLee0903** [speechbrain/speechbrain#1492]
Hi, the speechbrain team, 

Thanks for sharing this awesome tool to bring the convenience of reproducibility! I noticed that the input that feeds to train the discriminator is not the detached version. This means the gradients in the last step of the discriminator training process will not be cleared and thus perhaps affect the generator training process. 
Moreover, during the generator training process, the gradients of the discriminator will be accumulated since only the gradients of the generator will be cleaned up.

Therefore, I tested the gradients of the discriminator before executing line 324 of train.py, and then I just found they were not zero. I guess this part might be an issue since the accumulated gradients of the discriminator would affect the backpropagation of the generator. Maybe you should call self.d_optimizer.zero_grad() before executing loss.backward() (line 324).

Best regards,
CHI-CHANG LEE.

**Comment by ChangLee0903** [comment]
Sorry, I just read it wrong. No problems remain now.
